scripting/include/srccoop/entitypatch.inc

```diff
@@ -1611,3 +1611,76 @@ public MRESReturn Hook_MusicTrackAceptInput(int _this, Handle hReturn, Handle hP
 	return MRES_Ignored;
 }
 
+//------------------------------------------------------
+// func_trackautochange
+// Prevent track from being blocked by players and forced to go back down
+//------------------------------------------------------
+public MRESReturn Hook_TrackChangeBlocked(int _this, Handle hReturn, Handle hParams)
+{
+	if (!DHookIsNullParam(hParams, 1))
+	{
+		CBasePlayer pPlayer = DHookGetParam(hParams, 1);
+		if (pPlayer.IsValid())
+		{
+			pPlayer.Suicide();
+			
+			DHookSetReturn(hReturn, false);
+			return MRES_Supercede;
+		}
+	}
+	
+	return MRES_Ignored;
+}
+
+//------------------------------------------------------
+// func_tracktrain
+// Prevent trains from being blocked by players
+// Moves players up if not blocked, otherwise kills blocking player
+//------------------------------------------------------
+public MRESReturn Hook_TrackTrainBlocked(int _this, Handle hReturn, Handle hParams)
+{
+	static float vecOriginAdjusted[3];
+	
+	if (!DHookIsNullParam(hParams, 1))
+	{
+		CBasePlayer pPlayer = DHookGetParam(hParams, 1);
+		if (pPlayer.IsValid())
+		{
+			pPlayer.GetAbsOrigin(vecOriginAdjusted);
+			vecOriginAdjusted[2] += 64.0;
+			if (TR_PointOutsideWorld(vecOriginAdjusted))
+			{
+				pPlayer.Suicide();
+				DHookSetReturn(hReturn, false);
+				return MRES_Supercede;
+			}
+			
+			// Move up 2 if blocking and not out of world
+			vecOriginAdjusted[2] -= 62.0;
+			pPlayer.Teleport(vecOriginAdjusted);
+		}
+	}
+	
+	return MRES_Ignored;
+}
+
+//------------------------------------------------------
+// item_suit
+// play admire hand animation
+//------------------------------------------------------
+public Action Hook_SuitTouchPickup(const char[] output, int caller, int activator, float delay)
+{
+	static char szName[32];
+	CBaseEntity(caller).GetTargetname(szName, sizeof(szName));
+	// Only run this if it is a named item_suit
```

Comment:
There is a spawnflag for item_suit (131072) called Boot sequence. May be better to check it instead?

Reply:
I suppose it could, but that spawnflag doesn't exist in any other game, only Black Mesa.